src/app/static/src/app/apiService.ts

```diff
@@ -4,44 +4,95 @@ import {
     Failure,
     InitialiseModelRunResponse,
     ModelRunResultResponse,
-    Success,
     ValidateInputResponse
 } from "./generated";
+import {Commit} from "vuex";
 
-type ResponseData = ValidateInputResponse | ModelRunResultResponse | InitialiseModelRunResponse | InternalResponse | null
+type ResponseData =
+    ValidateInputResponse
+    | ModelRunResultResponse
+    | InitialiseModelRunResponse
+    | InternalResponse
+    | Boolean
 
-export class APIService {
+declare var appUrl: string;
 
-    get<T extends ResponseData>(url: string) {
-        return axios.get(url)
-            .then((response: AxiosResponse<Success>) => {
-                const payload = response && response.data;
-                return payload.data as T
-            })
-            .catch((e: AxiosError<Failure>) => {
-                return this.handleError(e)
-            })
-    }
+export interface API {
+
+    commitFirstErrorAsType: (commit: Commit, type: string) => API
```

Comment:
If the aim is terseness, maybe we should drop the 'First' from this method name - I think we can take it as read that you'll get the first error only, and this method is going to be invoked lots of times.